pkg/segment/writer/segstore.go

```diff
@@ -831,21 +831,31 @@ func CleanupUnrotatedSegment(segstore *SegStore, streamId string, resetSegstore
 	return nil
 }
 
-func (segstore *SegStore) isAnyAtreeColAboveCardLimit() (string, bool, uint64) {
+func (segstore *SegStore) getColsAboveCardLimit() ([]string, []uint64) {
 
-	for _, cname := range segstore.stbHolder.stbPtr.GetGroupByKeys() {
+	highCardCols := []string{}
+	highCardVals := []uint64{}
```

Comment:
nit: These are tightly coupled—they should always have the same length. Would it be better to combine them into a `map[string]uint64`? If not, what about combining them into a `[]someNewType` where `someNewType` has fields for the column name and cardinality?

Reply:
Have modified the code to take in columns to drop as an array of strings. If we create someNewType we would need to pass it to the agileTree and it won't be modular. I can however create a map, just let me know if this needs change as per latest code.

Reply:
Ok, I think it would be good to combine these into one map, but I haven't thought too much about this. So use your judgement about whether that will make the code cleaner.

Reply:
addressed in the latest commit. Used the `map[string]uint64`, have not used generics on the DropColumns method, because if we want to optimize further in future regarding some estimation of the order of dropping columns based on cardinality, it would be good to have it here and currently this function is explicitly used for dropping high cardinality columns.